pkg/kubecost/allocation.go

```diff
@@ -2131,34 +2131,63 @@ func (asr *AllocationSetRange) NewAccumulation() (*AllocationSet, error) {
 	return allocSet, nil
 }
 
-// AccumulateBy sums AllocationSets based on the resolution given. The resolution given is subject to the scale used for the AllocationSets.
-// Resolutions not evenly divisible by the AllocationSetRange window durations Accumulate sets until a sum greater than or equal to the resolution is met,
-// at which point AccumulateBy will start summing from 0 until the requested resolution is met again.
-// If the requested resolution is smaller than the window of an AllocationSet then the resolution will default to the duration of a set.
-// Resolutions larger than the duration of the entire AllocationSetRange will default to the duration of the range.
-func (asr *AllocationSetRange) AccumulateBy(resolution time.Duration) (*AllocationSetRange, error) {
-	allocSetRange := NewAllocationSetRange()
-	var allocSet *AllocationSet
-	var err error
+// AccumulateBy sums AllocationSets based on the AccumulateByType (calendar week or calendar month).
+// The accumulated set is determined by the start of the window of the allocation set.
+func (asr *AllocationSetRange) AccumulateBy(accumulateBy AccumulateByType) (*AllocationSetRange, error) {
+	if accumulateBy == AccumulateByNone {
+		return asr, nil
+	}
 
-	for i, as := range asr.Allocations {
-		allocSet, err = allocSet.Accumulate(as)
-		if err != nil {
-			return nil, err
+	if accumulateBy == AccumulateByWeek {
+		var toAccumulate *AllocationSetRange
+		result := NewAllocationSetRange()
+		for i, as := range asr.Allocations {
+			dayOfWeek := as.Window.Start().Weekday()
+
+			if toAccumulate == nil {
+				toAccumulate = NewAllocationSetRange()
+			}
+
+			toAccumulate.Append(as)
+
+			// current assumption is the week always ends on Saturday
+			if dayOfWeek == time.Saturday || i == len(asr.Allocations)-1 {
```

Comment:
So this is, "If we just finished the last day of the week, OR we just finished the last set in the range." Makes sense to me.